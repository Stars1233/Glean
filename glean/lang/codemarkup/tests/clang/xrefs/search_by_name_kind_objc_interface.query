query: search.code.SearchByNameKindAndLanguage { name = "J", language = Cpp, kinds = { just = Interface } }
perf: true
transform: [gensort, []]
