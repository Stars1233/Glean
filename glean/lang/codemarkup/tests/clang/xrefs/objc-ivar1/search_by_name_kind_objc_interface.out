[
  "@generated",
  {
    "key": {
      "name": "J",
      "language": 0,
      "kinds": 12,
      "entity": {
        "cxx": {
          "defn": {
            "objcContainer": {
              "key": {
                "declaration": {
                  "key": {
                    "id": { "interface_": { "key": "J" } },
                    "source": {
                      "file": { "key": "test.m" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 15,
                      "columnEnd": 4
                    }
                  }
                },
                "protocols": [ ],
                "members": {
                  "key": [
                    {
                      "variable": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "_i" },
                              "scope": { "global_": { } }
                            }
                          },
                          "type": { "key": "int" },
                          "kind": { "ivar": { "synthesize": false } },
                          "source": {
                            "file": { "key": "test.m" },
                            "lineBegin": 10,
                            "columnBegin": 3,
                            "lineEnd": 10,
                            "columnEnd": 8
                          }
                        }
                      }
                    },
                    {
                      "objcProperty": {
                        "key": {
                          "name": { "key": "i" },
                          "container": { "interface_": { "key": "J" } },
                          "type": { "key": "int" },
                          "isInstance": true,
                          "isOptional": false,
                          "isReadOnly": false,
                          "isAtomic": false,
                          "source": {
                            "file": { "key": "test.m" },
                            "lineBegin": 13,
                            "columnBegin": 1,
                            "lineEnd": 13,
                            "columnEnd": 35
                          }
                        }
                      }
                    },
                    {
                      "objcMethod": {
                        "key": {
                          "selector": { "key": [ "i" ] },
                          "locations": [ ],
                          "container": { "interface_": { "key": "J" } },
                          "signature": {
                            "key": {
                              "returns": { "key": "int" },
                              "parameters": [ ]
                            }
                          },
                          "isInstance": true,
                          "isOptional": false,
                          "isAccessor": true,
                          "source": {
                            "file": { "key": "test.m" },
                            "lineBegin": 13,
                            "columnBegin": 35,
                            "lineEnd": 13,
                            "columnEnd": 35
                          }
                        }
                      }
                    },
                    {
                      "objcMethod": {
                        "key": {
                          "selector": { "key": [ "setI" ] },
                          "locations": [ ],
                          "container": { "interface_": { "key": "J" } },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "i" },
                                  "type": { "key": "int" }
                                }
                              ]
                            }
                          },
                          "isInstance": true,
                          "isOptional": false,
                          "isAccessor": true,
                          "source": {
                            "file": { "key": "test.m" },
                            "lineBegin": 13,
                            "columnBegin": 35,
                            "lineEnd": 13,
                            "columnEnd": 35
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  }
]